### PR TITLE
WinForms DataGridView sample

### PR DESCRIPTION
**Pull request by mjrousos**
.NET Core 3 WinForms sample demonstrating a variety of DataGridView scenarios. Ported from https://code.msdn.microsoft.com/windowsdesktop/CSWinFormDataGridView-29783221

### CODE REVIEW

File: windowsforms/datagridview/CSWinFormDataGridView/CustomDataGridViewColumn/MainForm.cs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+* Copyright (c) Microsoft Corporation.
+* 
+* This sample demonstrates the use of custom column definitions within the Windows Forms 
+* DataGridView control.
+* 
+* The Employee ID, SSN, State and Zip Code columns use MaskedTextBox controls for format 
+* and validate their input.
+* 
+* This source is subject to the Microsoft Public License.
+* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
```
Comment (mjrousos) [inline windowsforms/datagridview/CSWinFormDataGridView/CustomDataGridViewColumn/MainForm.cs]: Just noticed we still reference MSLPL here. I'll remove this.

File: windowsforms/datagridview/CSWinFormDataGridView/ResourceDeserializationExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace CSWinFormDataGridView
+{
+    public static class ResourceDeserializationExtensions
+    {
+        static BinaryFormatter Formatter = new BinaryFormatter();
+
+        /// <summary>
+        /// Returns an image deserialized from a string resource to work around https://github.com/dotnet/corefx/issues/26745
+        /// </summary>
```
Comment (mjrousos) [inline windowsforms/datagridview/CSWinFormDataGridView/ResourceDeserializationExtensions.cs]: Now that there's a fix for dotnet/corefx#26745 in the works, I can probably remove this.

File: windowsforms/datagridview/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# DataGridView Samples
+This Windows Forms sample was ported to .NET Core from 
+[a previous .NET 4.5 sample](https://code.msdn.microsoft.com/windowsdesktop/CSWinFormDataGridView-29783221).
```
[the remark this answers is not in the document]
Comment (mjrousos) [reply windowsforms/datagridview/README.md]: Fixed

File: windowsforms/datagridview/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+* DataGridViewPaging
+* EditingControlHosting
+* JustInTimeDataLoading
+* MultipleLayeredColumnHeader
+
+Note that the DataGridViewPaging and JustInTimeDataLoading samples use the 
+NorthWind sample database (in order to have a large amount of data to display 
+in the grid view). Before running these scenarios, follow the 
+[these instructions](https://docs.microsoft.com/dotnet/framework/data/adonet/sql/linq/downloading-sample-databases) 
+to setup the NorthView sample database and make sure that the connection
```
[the remark this answers is not in the document]
Comment (mjrousos) [reply windowsforms/datagridview/README.md]: Fixed

File: windowsforms/datagridview/CSWinFormDataGridView/JustInTimeDataLoading/ReadMe.txt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+4.  Handle the CellValueNeeded event to retrieve the requested cell value 
+    from the data store or the Customer object currently in edit. 
+
+
+/////////////////////////////////////////////////////////////////////////////
+References:
+
+1. Implementing Virtual Mode with Just-In-Time Data Loading in the Windows 
+   Forms DataGridView Control
+   http://msdn.microsoft.com/en-us/library/ms171624.aspx
```
[the remark this answers is not in the document]
Comment (mjrousos) [reply windowsforms/datagridview/CSWinFormDataGridView/JustInTimeDataLoading/ReadMe.txt]: Fixed; thanks!

File: windowsforms/datagridview/CSWinFormDataGridView/MultipleLayeredColumnHeader/ReadMe.txt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    cell;
+
+5.  Handle the DataGridView.Paint event to draw "merged" header cells;
+
+
+/////////////////////////////////////////////////////////////////////////////
+References:
+
+1. DataGridView Class
+   http://msdn.microsoft.com/en-us/library/system.windows.forms.datagridview.aspx
```
[the remark this answers is not in the document]
Comment (mjrousos) [reply windowsforms/datagridview/CSWinFormDataGridView/MultipleLayeredColumnHeader/ReadMe.txt]: Fixed.

File: windowsforms/datagridview/CSWinFormDataGridView/EditingControlHosting/ReadMe.txt [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+7. Handle the EditingControlShowing event to pass the focus to the MaskedTextBox
+   when begin editing with keystrokes;
+
+
+/////////////////////////////////////////////////////////////////////////////
+References:
+
+1. DataGridView Class
+   http://msdn.microsoft.com/en-us/library/system.windows.forms.datagridview.aspx
```
[the remark this answers is not in the document]
Comment (mjrousos) [reply windowsforms/datagridview/CSWinFormDataGridView/EditingControlHosting/ReadMe.txt]: Fixed

**Comment by mjrousos** [comment]
@rpetrusha, I've updated the PR based on your comments. Any other suggestions here or is this one ready to merge?

**Comment by rpetrusha** [comment]
Thanks for updating your PR, @mjrousos. This is ready to merge.

**Pull request merged**